tools/spice/spice_utils.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import os
+import re
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Optional
+
+import spiceypy as spice
+from spiceypy.utils.exceptions import NotFoundError
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(__name__)
+
+
+class SpiceKernelManager:
```

Comment:
Looks like this might even already be implemented in spiceypy? So, do we need our own version here or can you expose that somehow instead.
https://github.com/AndrewAnnex/SpiceyPy/blob/22f7aaa0944d01994540f0acd5136a54ed239efe/src/spiceypy/spiceypy.py#L289
https://spiceypy.readthedocs.io/en/main/contextmanager.html